File: Prosary/Models/PrayerReminder.cs
namespace Prosary.Models;

/// <summary>A single daily reminder time attached to a <see cref="Prayer"/>. Scheduled via
/// <c>WindowsReminderScheduler</c>.</summary>
public sealed record PrayerReminder
{
    public Guid Id { get; init; } = Guid.NewGuid();

    /// <summary>0–23.</summary>
    public int Hour { get; init; }

    /// <summary>0–59.</summary>
    public int Minute { get; init; }

    public bool IsEnabled { get; init; } = true;

    public PrayerReminder() { }

    public PrayerReminder(int hour, int minute = 0, bool isEnabled = true)
    {
        Hour = hour;
        Minute = minute;
        IsEnabled = isEnabled;
    }

    public string DisplayTime => AsDate.ToString("t");

    /// <summary>A <see cref="DateTime"/> whose time-of-day matches this reminder, for use with a
    /// <c>TimePicker</c>.</summary>
    public DateTime AsDate => DateTime.Today.AddHours(Hour).AddMinutes(Minute);
}

File: Prosary/Models/PrayerReminder.swift

//
//  PrayerReminder.swift
//  Prosary
//

import Foundation

struct PrayerReminder: Hashable, Codable, Identifiable {
  var id = UUID()
  var hour: Int     // 0–23
  var minute: Int   // 0–59
  var isEnabled: Bool = true

  init(hour: Int, minute: Int = 0, isEnabled: Bool = true) {
    self.hour = hour
    self.minute = minute
    self.isEnabled = isEnabled
  }

  var displayTime: String {
    let comps = DateComponents(hour: hour, minute: minute)
    guard let date = Calendar.current.date(from: comps) else {
      return String(format: "%02d:%02d", hour, minute)
    }
    return date.formatted(date: .omitted, time: .shortened)
  }

  /// A `Date` whose time component matches this reminder, for use with `DatePicker`.
  var asDate: Date {
    Calendar.current.date(from: DateComponents(hour: hour, minute: minute)) ?? Date()
  }
}
